src/snakemake/persistence.py

```
@@ -631,6 +636,21 @@ def _params(self, job: Job):
             )
         )
 
+    @lru_cache()
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)